datafusion/sql/src/parser.rs

```diff
@@ -41,10 +41,7 @@ fn parse_file_type(s: &str) -> Result<FileType, ParserError> {
         "NDJSON" => Ok(FileType::NdJson),
         "CSV" => Ok(FileType::CSV),
         "AVRO" => Ok(FileType::Avro),
-        other => Err(ParserError::ParserError(format!(
-            "expect one of PARQUET, AVRO, NDJSON, or CSV, found: {}",
-            other
-        ))),
+        name => Ok(FileType::Custom(name.to_string()))
```

Comment:
Eventually I'd like to un-hard-code the built-in tables and just have a `ListingTableFactory` in the map so all these behave the same.

Reply:
Also, I'd love to use the postgres `with options` syntax to pass an arbitrary hashmap for credentials, host & port, etc.

https://www.postgresql.org/docs/current/sql-createforeigntable.html

```
CREATE FOREIGN TABLE [ IF NOT EXISTS ] table_name ( [
  { column_name data_type [ OPTIONS ( option 'value' [, ... ] ) ]
  ```